modules/extensions-support/src/main/java/org/mule/module/extension/DescriberSupport.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.module.extension;
+
+import org.mule.extension.introspection.Describer;
+import org.mule.extension.introspection.declaration.Construct;
+import org.mule.module.extension.internal.introspection.AnnotationsBasedDescriber;
+
+/**
+ * Utility class for easily building a {@link Describer}
+ * that is discoverable through SPI for an extension defined
+ * through annotations without coupling
+ * to mule's mechanism to actually parse those annotations.
+ * <p/>
+ * Users simply extend this class and add a default constructor
+ * which delegates into {@link #DescriberSupport(Class)}.
+ *
+ * @since 3.7.0
+ */
+public abstract class DescriberSupport implements Describer
```

Comment:
I don't agree with this. AbstractDescriber is a misleading name because it makes you think that it's intended to be the base class of every Describer or an implementation of the  template method pattern.

Instead, this is just to help extension developers to build the describer. It delegates into AnnotationsBasedDescriber into a one and only final method. It does not have abstract methods, it just has state. 

This is a support class, not an abstract implementation.